### DISCUSSION TITLE
Protect pb collection

### DISCUSSION THREAD

**Discussion by thewebguy27** [pocketbase/pocketbase#3848] [Q&A]
Hi I need to protect my collections data so that it could not be accessed directly using a URL 

"/api/collections/collectionIdOrName/records"

and only allow the users that are registered to access the data I tried using "@request.auth.id != ''" but it did not work

**Comment by dschissler** [comment]
There is a list and view filter rule.  Make sure that you're not signed in already.  Try another browser or private browser mode.

**Reply by thewebguy27** [reply]
I did use the private browser mode but still after signing in the collection record is shown empty using the above mentioned rule

**Reply by ganigeorgiev** [reply]
@thewebguy27 From https://pocketbase.io/docs/api-rules-and-filters/:

> PocketBase API Rules act also as records filter!
Or in other words, you could for example allow listing only the "active" records of your collection, by using a simple filter expression such as: status = "active" (where "status" is a field defined in your Collection).
>
> **Because of the above, the API will return 200 empty items response in case a request doesn't satisfy a listRule, 400 for unsatisfied createRule and 404 for unsatisfied viewRule, updateRule and deleteRule.
All rules will return 403 in case they were "locked" (aka. admin only) and the request client is not an admin.**
>
> The API Rules are ignored when the action is performed by an authorized admin (admins can access everything)!
